File: -- src_R/| |-- 01_analisis_teks.R
#Program 1: Analisis Teks - Menghitung Jumlah Kata dan Jumlah Kalimat
hitung_kata <- function(teks) {
  #Memisahkan teks berdasarkan spasi lalu menghitung elemen
  kata <- unlist(strsplit(trimws(teks), "\\s+"))
  return(length(kata))
}
hitung_kalimat <- function(teks) {
  
  #Memisahkan teks berdasarkan tanda titik, lalu hitung bagian tidak kosong
  bagian <- strsplit(teks, "\\.")[[1]]
  jumlah <- length(bagian[trimws(bagian) != ""])
  return(jumlah)
}

#Input dari pengguna
teks <- readline(prompt = "Masukkan teks: ")

#Proses perhitungan
jumlah_kata <- hitung_kata(teks)
jumlah_kalimat <- hitung_kalimat(teks)

#Output hasil
print(paste0("Teks tersebut memuat ", jumlah_kalimat, " kalimat dan ", jumlah_kata, " kata."))

File: -- src_python/| |-- 01_analisis_teks.py
#Menghitung jumlah kata dan jumlah kalimat
#Memisahkan teks berdasarkan spasi
def hitung_kata(teks):
   kata = teks.split()
   return len(kata)
#Memisahkan teks berdasarkan tanda titik
def hitung_kalimat(teks):
   bagian = teks.split(".")
   jumlah = len([b for b in bagian if b.strip() != ""]) #Menyaring bagian kosong
   return jumlah
#Input dari pengguna
teks = input("Masukkan teks: ")
#Proses perhitungan
jumlah_kata = hitung_kata(teks)
jumlah_kalimat = hitung_kalimat(teks)
#Output hasil
print(f"Teks tersebut memuat {jumlah_kalimat} kalimat dan {jumlah_kata} kata.")
